operator/controllers/manifest_controller.go

```diff
@@ -202,9 +202,58 @@ func (r *ManifestReconciler) HandleCharts(chart v1alpha1.ChartInfo, logger logr.
 	return nil
 }
 
+func (r *ManifestReconciler) ResponseHandlerFunc(ctx context.Context, chartCount int, logger *logr.Logger, namespacedName client.ObjectKey) {
+	endState := v1alpha1.ManifestStateReady
+	var err error
+	for a := 1; a <= chartCount; a++ {
+		err = <-r.ResponseChan
+		if err != nil {
+			logger.Error(err, "chart installation failure!!!")
+			endState = v1alpha1.ManifestStateError
+			break
+		}
+	}
+
+	latestManifestObj := &v1alpha1.Manifest{}
+	if err := r.Get(ctx, namespacedName, latestManifestObj); client.IgnoreNotFound(err) != nil {
```

Comment:
Two questions:

- What happens if "NotFound"? The rest of this code works?
- What happens if another error occurs? The rest of this code works?

Reply:
It's probably a good idea to `return` here